### PR TITLE
chore: Update dependencies (again)

### PR DESCRIPTION
**Pull request by alamb**
Rationale: Gotta keep pace with the pace of DataFusion changes in Arrow!

But more to the point, as I see changes happen to DataFusion that will affect us, I try to update the deps asap so I avoid a massive and painful upgrade in the future

### CODE REVIEW

File: mutable_buffer/src/table.rs [hunk 1]
```diff
@@ -435,15 +435,15 @@ impl Table {
         sort_exprs.push(TIME_COLUMN_NAME.into_sort_expr());
 
         // Order by
-        let plan_builder = plan_builder.sort(sort_exprs).context(BuildingPlan)?;
+        let plan_builder = plan_builder.sort(&sort_exprs).context(BuildingPlan)?;
```
Comment (alamb) [inline mutable_buffer/src/table.rs:438]: I am not convinced this is an improvement (as it just forces the user of the builder) to copy the exprs and if the caller has created them (as here) they can't be used. 

I am following up with the DataFusion folks: https://github.com/apache/arrow/pull/9338#pullrequestreview-580776979

**Comment by alamb** [comment]
cc @tustvold

File: mutable_buffer/src/table.rs [hunk 2]
```diff
@@ -377,7 +377,7 @@ impl Table {
         let select_exprs = vec![col(column_name)];
 
         plan_builder
-            .project(select_exprs)
+            .project(&select_exprs)
```
Comment (Dandandan) [inline mutable_buffer/src/table.rs:380]: `select_exprs` doesn't have to be a `Vec` anymore `&[col(column_name)]`.

Comment (alamb) [reply mutable_buffer/src/table.rs:380]: 👍  it is a good point. I may wait to see what, if any, further changes are made to the plan builder API before changing this callsite. Thanks for taking a look @Dandandan

**Review by e-dard — APPROVED** [review]
LGTM 👍

**Pull request merged**